File: app/models/cloze_notes_concept.rb
# Anki Books, a note-taking app to organize knowledge,
# is licensed under the GNU Affero General Public License, version 3
# Copyright (C) 2023 Kyle Rego

# frozen_string_literal: true

# == Schema Information
#
# Table name: cloze_notes_concepts
#
#  id            :uuid             not null, primary key
#  cloze_note_id :uuid             not null
#  concept_id    :uuid             not null
#  created_at    :datetime         not null
#  updated_at    :datetime         not null
#
# Foreign Keys
#
#  fk_rails_...  (cloze_note_id => cloze_notes.id)
#  fk_rails_...  (concept_id => concepts.id)
#
class ClozeNotesConcept < ApplicationRecord
  belongs_to :cloze_note
  belongs_to :concept
end

File: AnkiBooks/Models/ClozeNotesConcept.cs
using System;
using System.Collections.Generic;

namespace AnkiBooks.Models;

public partial class ClozeNotesConcept
{
    public Guid Id { get; set; }

    public Guid ClozeNoteId { get; set; }

    public Guid ConceptId { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual ClozeNote ClozeNote { get; set; } = null!;

    public virtual Concept Concept { get; set; } = null!;
}
